src/main/java/seedu/address/logic/commands/MarkOrderCommand.java

```diff
@@ -40,7 +40,7 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Order orderToMark = lastShownList.get(targetIndex.getZeroBased());
-        model.markOrder(orderToMark);
+        model.markOrder(orderToMark, targetIndex.getZeroBased());
```

Comment:
I think there might be a potential bug here? It looks to me like you parse the index the user gives directly for the marktask/markorder command.

However, if the user filters the list, e.g. they type findtask zz and there are now 0 tasks listed, marktask 1 should fail, even if the tasklist is nonempty.